imports/ui/pages/overview/components/dashboardModal.js

```diff
@@ -33,24 +38,104 @@ Template.dashboardModal.helpers({
   },
   dashboardURL() {
     const dashboardId = Template.instance().dashboardId.get()
-    if (dashboardId) {
-      return FlowRouter.url('dashboard', { _id: dashboardId })
+    if (dashboardId.slug){
+      return FlowRouter.url('dashboard', { _id: dashboardId.slug })
+    }
+    if (dashboardId._id) {
+      return FlowRouter.url('dashboard', { _id: dashboardId._id })
     }
     return ''
   },
+  slugStatus() {
+    return Template.instance().slugAvailable.get();
+  },
+  slugSanitzed() {
+    return Template.instance().slugSanitized.get();
+  },
+  slugStatusOn() {
+    if (Template.instance().slugAvailable.get() != null) {
+      return true
+    } else {
+      return false
+    }
+  },
+  passwordWarningOn() {
+    return !Template.instance().passwordInserted.get()
+  },
 })
 
 Template.dashboardModal.events({
   'click .js-create-dashboard': (event, templateInstance) => {
     event.preventDefault()
+    let startDate, endDate
+    const period = document.getElementById("dashboardPeriod")?.value
+    if (period === 'custom'){
+      startDate = document.getElementById("customStartDate")?.value
+      endDate = document.getElementById("customEndDate")?.value
+    } else {
+      startDate = 'N/A'
+      endDate = 'N/A'
+    }
+    const password = document.getElementById("dashboard-password")?.value
+    const slug = document.getElementById("dashboard-slug")?.value
+
     Meteor.call('addDashboard', {
-      projectId: templateInstance.data.projectId.get(), resourceId: 'all', customer: 'all', timePeriod: templateInstance.$('#period').val(),
+      projectId: templateInstance.data.projectId.get(), timePeriod: period, startDate : startDate, endDate: endDate, password : password,slug:slug
```

Comment:
see previous comments, please keep resourceId and customer for backwards compatibility if possible